### ISSUE TITLE
verdi calculation kill hangs forever

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#1522]
I have an old calculation that remained in state `NEW` (it's possible that it was generated with a pre-1.0 version of the code):
```
224322  91D ago     Unknown | None  quantumespresso.pw  localhost   NEW | None
```
If I try to kill it with verdi calculation kill, after asking for confirmation it hangs indefinitely. If I CTRL+C, I get this traceback (maybe it helps):
```
Traceback (most recent call last):
  File "/home/pizzi/.virtualenvs/aiida_core/bin/verdi", line 9, in <module>
    sys.exit(run())
  File "/home/pizzi/git/aiida_core/aiida/cmdline/verdilib.py", line 1056, in run
    aiida.cmdline.verdilib.exec_from_cmdline(sys.argv)
  File "/home/pizzi/git/aiida_core/aiida/cmdline/verdilib.py", line 1041, in exec_from_cmdline
    CommandClass.run(*argv[command_position + 1:])
  File "/home/pizzi/git/aiida_core/aiida/cmdline/baseclass.py", line 234, in run
    function_to_call(*args[1:])
  File "/home/pizzi/git/aiida_core/aiida/cmdline/commands/calculation.py", line 694, in calculation_kill
    print('Calculation<{}> killing failed {}'.format(future[0], result))
  File "/home/pizzi/git/aiida_core/aiida/work/rmq.py", line 240, in __exit__
    self.close()
  File "/home/pizzi/git/aiida_core/aiida/work/rmq.py", line 243, in close
    self._communicator.disconnect()
  File "/home/pizzi/.virtualenvs/aiida_core/local/lib/python2.7/site-packages/kiwipy/rmq/communicator.py", line 302, in disconnect
    self._loop.run_sync(self._message_publisher.disconnect)
  File "/home/pizzi/.virtualenvs/aiida_core/local/lib/python2.7/site-packages/tornado/ioloop.py", line 453, in run_sync
    self.start()
  File "/home/pizzi/.virtualenvs/aiida_core/local/lib/python2.7/site-packages/tornado/ioloop.py", line 755, in start
    raise RuntimeError("IOLoop is already running")
RuntimeError: IOLoop is already running
```

I don't know if this happens only for old calcs (e.g. because they miss some attribute) or for any calc in the NEW state or anyway not yet submitted.

**Comment by giovannipizzi** [comment]
I just realised that the same behaviour happens also for any other calc - I tried this:
```
224335  1m ago      Waiting | None  quantumespresso.pw  daint-mc    WITHSCHEDULER | QUEUED
```

**Comment by sphuber** [comment]
The new `kill` implementation always goes over RabbitMQ, that is to say the command fires a request to all subscribers that have that process to kill it, except that no one is listening. The only one that could have it is the daemon, but apparently it does not have it. Since no one is responding the call "hangs". This can happen when a calculation gets submitted and something goes wrong before a checkpoint can be created by the daemon, for example during a migration and an inconsistent state. There is currently no real way to kill it through the API other than set the appropriate attributes in the shell manually

**Comment by sphuber** [comment]
There is two issues here that should be addressed:

* Have a proper timeout and response if the RabbitMQ request is not handled by the daemon
* Have a last resort "clean-up" version of kill that does not go over RabbitMQ but just directly kills a node in the database. This should only be done for "lost" processes

**Comment by giovannipizzi** [comment]
Indeed, I think from the user point of view it does not matter if the calculation is with the scheduler or not, when they run 'kill' they just want it to disappear (and make sure it does not run).
Anyway now I have problems also for new calculations properly submitted through the daemon and being managed by it (see my previous comment (unless it's excepted that it takes tens of seconds).

**Comment by sphuber** [comment]
The only tricky thing might be how to determine whether we get a timeout because the process is not with any runner, or because a runner *is* actually running it but could not respond in time. I do not think we can tell the difference for sure. It would be pretty bad to go ahead and kill it directly in the database and it turns out it was still being run by some runner

**Comment by sphuber** [comment]
This is a duplicate of #1824 and #1712

**Comment by sphuber** [comment]
Fixed by proxy in #1860 
The `verdi process` commands now have a proper timeout so wont hang forever.
Dealing with unreachable processes will be dealt with in issue #1712
